### PR TITLE
Server unit tests

### PR DESCRIPTION
**Pull request by JoshVanL**
Adds unit tests to Server.

Branched from #74  

Adds an interface to certmanager for faking.

/assign @irbekrm 
/hold

### CODE REVIEW

File: pkg/certmanager/certmanager.go
Comment (irbekrm) [inline pkg/certmanager/certmanager.go:53]: Suggestion: call this `Signer` rather than `Interface`

**Review by irbekrm — CHANGES_REQUESTED** [review]
The tests look great to me!
I've added an optional suggestion, otherwise happy to lgtm once #74 gets merged and this is rebased.

File: pkg/certmanager/certmanager.go
Comment (JoshVanL) [reply pkg/certmanager/certmanager.go:53]: Thanks @irbekrm. 

This is actually an intentional change. I've come around to the idea that `signer.Signer` is bad because it is stuttering, whereas `signer.Interface` makes sense and doesn't stutter.

Comment (irbekrm) [reply pkg/certmanager/certmanager.go:53]: I guess `certmanager.Signer` would not be a stutter and may be expressing the intent more clearly, but `Interface` is ok too imo

Comment (JoshVanL) [reply pkg/certmanager/certmanager.go:53]: This is true, I will change it to ‘Signer’.

**Comment by JoshVanL** [comment]
Thanks @irbekrm, this is good for another review.

/hold cancel

File: pkg/certmanager/certmanager.go
Comment (JoshVanL) [reply pkg/certmanager/certmanager.go]: Done in https://github.com/cert-manager/istio-csr/pull/75/commits/8550a4c187ff733d02423553413fdbefba4cdce8

**Comment by irbekrm** [comment]
@JoshVanL was this PR not supposed to have another commit with the server tests and the fake?

**Comment by JoshVanL** [comment]
> @JoshVanL was this PR not supposed to have another commit with the server tests and the fake?

Not sure what's happened, they've managed to be apart of [this PR](https://github.com/cert-manager/istio-csr/pull/74) now. Strange.

**Comment by irbekrm** [comment]
Ahh cool, so this is just the rename.
That looks good to me :) 

/lgtm

**Pull request merged**
